src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty() || !userByEmail.get().getPassword()
+                .equals(HashUtil.hashPassword(password, userByEmail.get().getSalt()))) {
+            throw new AuthenticationException("There is no user with this email or password.");
+        }
+        return userByEmail.get();
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("User with this email address "
+                    + "already exists: " + email);
+        }
+        if (!email.contains("@")) {
+            throw new RegistrationException("Incorrect email address");
+        }
```

Comment:
let's remove this check, in one of the next tasks you will be implementing validation logic for email